src/js/utils.js

```diff
@@ -24,20 +23,14 @@ exports.parseQueryParams = () => {
   }, {});
 };
 
-/**
- * Creates an object based on a form's current input values.
- *
- * @param {object} form
- * @returns {object}
- */
-exports.parseForm = (form) => {
-  const filter = {};
-  const elements = form.querySelectorAll('input[type="checkbox"]:checked');
-  Array.from(elements).forEach((element) => {
-    if (!filter[element.name]) {
-      filter[element.name] = [];
-    }
-    filter[element.name].push(element.value);
-  });
-  return filter;
+exports.decodeFormURLEncodedSpaces = (value) => {
+  /* Turn "foo+bar" into "foo bar".
+   * Context: decodeURIComponent (used above in parseQueryParams) does not
+   * convert `+` to spaces, because the `+` is from the spec for
+   * application/x-www-form-urlenconded, not encodeURIComponent)
+   *
+   * @param {string} value
+   * @returns {string}
+   */
+  return value.replace(/\+/g, ' ');
```

Comment:
Wrapping this in a func is heavy, i know, but i wanted to make it explicit, testable and easily extensible

Reply:
Equally, it's a minor pain to have to be doing this, but I think it's unavoidable. Some browsers (eg FF) use `+` for spaces, so we have to expect it